cns/ipampoolmonitor/ipampoolmonitor.go

```diff
@@ -15,57 +15,63 @@ import (
 
 const defaultMaxIPCount = int64(250)
 
-type CNSIPAMPoolMonitor struct {
-	MaximumFreeIps           int64
-	MinimumFreeIps           int64
-	cachedNNC                v1alpha.NodeNetworkConfig
+type PoolMon struct {
+	nnc                      *v1alpha.NodeNetworkConfig
```

Comment:
Why do we need to persist the whole nnc object (which can include upto 250 ips)?

Reply:
Just Scaler and Spec is enough, no?